src/components/ExerciseQuestion.tsx

```diff
@@ -0,0 +1,140 @@
+  incorrectAnswerId: ID;
+  answerId: ID;
+  attempts_remaining: number;
+  published_comments: string;
+  detailedSolution: string;
+  canAnswer: boolean;
+  needsSaved: boolean;
+  canUpdateCurrentStep: boolean;
+  attempt_number: number;
+  apiIsPending: boolean;
+  displaySolution: boolean;
+  available_points: availablePoints;
+  exercise_uid: string;
+  free_response?: string;
+}
+
+const AttemptsRemaining = ({ count }: { count: number }) => {
+  return (
+    <div>{count} attempt{count === 1 ? '' : 's'} left</div>
+  );
+}
+
+const PublishedComments = ({ published_comments }: { published_comments: string }) => {
+  if (!published_comments) { return null; }
+
+  return (
+    <div>
+      <strong>Feedback:</strong> {published_comments}
+    </div>
+  );
+}
+
+const SaveButton = (props: {
+  disabled: boolean, isWaiting: boolean, attempt_number: number
+} & React.ComponentPropsWithoutRef<'button'>) => (
+  <Button
+    {...props}
+    waitingText="Saving…"
+    isWaiting={props.isWaiting}
+    data-test-id="submit-answer-btn"
+  >
+    {props.attempt_number == 0 ? 'Submit' : 'Re-submit'}
+  </Button>
+);
+
+const NextButton = (props: {
+  canUpdateCurrentStep: boolean,
+  onNextStep: ExerciseQuestionProps['onNextStep']
+}) => {
+  return (
+    <Button onClick={props.onNextStep} data-test-id="continue-btn">
+      {props.canUpdateCurrentStep ? 'Continue' : 'Next'}
+    </Button>
+  );
+}
+
+const FreeResponseReview = ({ free_response }: Pick<ExerciseQuestionProps, "free_response">) => {
+  if (!free_response) { return null; }
+  return (
+    <>
+      <div className="free-response">{free_response}</div>
+    </>
+  );
+}
+
+export const ExerciseQuestion = (props: ExerciseQuestionProps) => {
+  const {
+    question, task, answer_id_order, onAnswerChange, feedback_html, correct_answer_feedback_html,
+    is_completed, correct_answer_id, incorrectAnswerId, choicesEnabled, questionNumber,
+    answerId, hasMultipleAttempts, attempts_remaining, published_comments, detailedSolution,
+    canAnswer, needsSaved, attempt_number, apiIsPending, onAnswerSave, onNextStep, canUpdateCurrentStep,
+    displaySolution, available_points, free_response
+  } = props;
+
+  return (
+    <div data-test-id="student-exercise-question">
+      <Question
+        task={task}
+        question={question}
+        answerIdOrder={answer_id_order}
+        choicesEnabled={choicesEnabled}
+        answer_id={answerId}
+        questionNumber={questionNumber}
+        onChange={onAnswerChange}
+        feedback_html={feedback_html}
+        correct_answer_feedback_html={correct_answer_feedback_html}
+        correct_answer_id={is_completed ? correct_answer_id : null}
+        incorrectAnswerId={incorrectAnswerId}
+        className="step-card-body"
+        hideAnswers={false}
+        displayFormats={false}
+        displaySolution={displaySolution}
+      >
+        <FreeResponseReview free_response={free_response} />
+      </Question>
+      <StepCardFooter>
```

Comment:
Should this component be imported from a separate file? And will the Save and Next buttons be used in assignable?

Reply:
hmm it looks like the StepCardFooter in the free response is a bit different so perhaps it's not worth it. Though I still wonder about all the different buttons.

Reply:
Maybe this becomes a single button that gets both callbacks and is controlled with a much simpler state prop, it would make the render cases much easier to reason about.

Reply:
I think that's a good idea because the free response also has a "revert" button